src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -475,11 +507,14 @@ void FancyZones::AddZoneWindow(HMONITOR monitor, PCWSTR deviceId) noexcept
     wil::unique_cotaskmem_string virtualDesktopId;
     if (SUCCEEDED_LOG(StringFromCLSID(m_currentVirtualDesktopId, &virtualDesktopId)))
     {
-        const bool flash = m_settings->GetSettings().zoneSetChange_flashZones;
+        const bool newVirtualDesktop = m_virtualDesktopIds[m_currentVirtualDesktopId];

```

Comment:
`m_virtualDesktopIds` is modified by `HandleVirtualDesktopUpdates` that runs on a different thread, so you can't read/write it here unless you lock it.
Also, since `HandleVirtualDesktopUpdates` may have already added the new desktop to `m_virtualDesktopIds`, this check may fail to detect a new virtual desktop, correct?

Reply:
Didn't see previous @yuyoyuppe comment about it, anyway saying it twice doesn't hurt ;)